### PR TITLE
feat: Payroll Entry customisations

### PR DESCRIPTION
**Pull request by ks093**
## Is this a Feature, Chore or Bug?
- [x] Feature
- [] Chore
- [] Bug

## Clearly and concisely describe the feature, chore or bug.
1. Adding custom fields for payroll cycle start and end date in Project, Employee
2. Property setter for payroll frequency
3. Custom fields in Payroll entry for Project configuration
4. When we create a new Payroll Entry, added a project filter: All External Projects or Specific project.
5. When all external projects is selected, the system will get all the employees assigned to external projects which have the same payroll cycle as the one selected in Payroll Entry.
6. When specific project is selected, it will fetch all the employees if project's payroll cycle matches the one selected in payroll entry.

## Output screenshots (optional)
image
image

## Is there a business logic within a doctype?
    - [x] Yes
    - [] No

## Did you test with the following dataset?
- [] Existing Data
- [] New Data

## Which browser(s) did you use for testing?
  - [x] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/fixtures/property_setter.json [hunk 1]
```diff
@@ -1,20 +1,4 @@
 [
- {
```
Comment (NoBoneZ) [inline one_fm/fixtures/property_setter.json:2]: You removed a lot of fields from here, was it intentional ?

Comment (ks093) [reply one_fm/fixtures/property_setter.json:2]: Yes. Refactoring as per the new process.

File: one_fm/overrides/payroll_entry.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+	Employee = frappe.qb.DocType("Employee")
+	Attendance = frappe.qb.DocType("Attendance")
+
+	query = (
+		frappe.qb.from_(Employee)
+		.join(SalaryStructureAssignment)
+		.on(Employee.name == SalaryStructureAssignment.employee)
+		.where(
+			(SalaryStructureAssignment.docstatus == 1)
+			& (Employee.status != "Inactive")
```
Comment (NoBoneZ) [inline one_fm/overrides/payroll_entry.py]: On the employee doctype, there is no inactive status

just active, courtcase, left, absconding

Comment (ks093) [reply one_fm/overrides/payroll_entry.py]: @AbrahamAdekunle Nice catch. Good work.

**Review by NoBoneZ — APPROVED** [review]

**Pull request merged**